scripts/speech_recognition/confidence/benchmark_asr_confidence.py

```diff
@@ -0,0 +1,427 @@
+        classpath = model_cfg.target  # original class path
+        imported_class = model_utils.import_class_by_path(classpath)  # type: ASRModel
+        logging.info(f"Restoring model : {imported_class.__name__}")
+        asr_model = imported_class.restore_from(
+            restore_path=cfg.model_path, map_location=map_location
+        )  # type: ASRModel
+    else:
+        # restore model by name
+        asr_model = ASRModel.from_pretrained(
+            model_name=cfg.pretrained_name, map_location=map_location
+        )  # type: ASRModel
+
+    trainer = pl.Trainer(devices=device, accelerator=accelerator)
+    asr_model.set_trainer(trainer)
+    asr_model = asr_model.eval()
+
+    # Check if ctc or rnnt model
+    is_rnnt = hasattr(asr_model, 'joint')
+
+    # Check that the model has the `change_decoding_strategy` method
+    if not hasattr(asr_model, 'change_decoding_strategy'):
+        raise RuntimeError("The asr_model you are using must have the `change_decoding_strategy` method.")
+
+    # get filenames and reference texts from manifest
+    filepaths = []
+    reference_texts = []
+    if os.stat(cfg.dataset_manifest).st_size == 0:
+        logging.error(f"The input dataset_manifest {cfg.dataset_manifest} is empty. Exiting!")
+        return None
+    manifest_dir = Path(cfg.dataset_manifest).parent
+    with open(cfg.dataset_manifest, 'r') as f:
+        for line in f:
+            item = json.loads(line)
+            audio_file = Path(item['audio_filepath'])
+            if not audio_file.is_file() and not audio_file.is_absolute():
+                audio_file = manifest_dir / audio_file
+            filepaths.append(str(audio_file.absolute()))
+            reference_texts.append(item['text'])
+
+    # setup AMP (optional)
+    if cfg.amp and torch.cuda.is_available() and hasattr(torch.cuda, 'amp') and hasattr(torch.cuda.amp, 'autocast'):
+        logging.info("AMP enabled!\n")
+        autocast = torch.cuda.amp.autocast
+    else:
+
+        @contextlib.contextmanager
+        def autocast():
+            yield
+
+    # do grig-based benchmarking if grid_params is provided, otherwise a regular one
+    work_dir = Path(cfg.output_dir)
+    os.makedirs(work_dir, exist_ok=True)
+    report_legend = (
+        ",".join(
+            [
+                "model_type",
+                "aggregation",
+                "blank",
+                "method_name",
+                "entropy_type",
+                "entropy_norm",
+                "temperature",
+                "target_level",
+                "auc_roc",
+                "auc_pr",
+                "auc_nt",
+                "nce",
+                "ece",
+                "auc_yc",
+                "max_yc",
+                "std_yc",
+            ]
+        )
+        + "\n"
+    )
+    model_typename = "RNNT" if is_rnnt else "CTC"
+    if cfg.grid_params:
+        report_file = work_dir / Path("report.csv")
```

Comment:
Line 357 and line 395 duplicate each other. Please move that line out of if/else scope to the line before line 356.